### PR TITLE
Always delete job post handling in worker

### PR DESCRIPTION
**Pull request by rtnpro**
even during success, failure or error.

### CODE REVIEW

File: container_pipeline/workers/base.py [hunk 1]
```diff
@@ -47,19 +47,27 @@ def export_logs(self, logs, destination):
...
+                # encountered in post delivering build report mails to user
+                dfh = DynamicFileHandler(self.logger, debug_logs_file)
+                self.logger.info('Got job: {}'.format(job))
+                try:
+                    self.handle_job(job)
+                except Exception as e:
+                    self.logger.error(
+                        'Error in handling job: {}\nJob details: {}'.format(
+                            e, job), extra={'locals': locals()}, exc_info=True)
+                dfh.remove()
```
Comment (bamachrn) [inline container_pipeline/workers/base.py]: Should we move this to finally also?

Comment (rtnpro) [inline container_pipeline/workers/base.py]: Nope. What if this errors out? We need to delete the job at any cost.

Comment (mohammedzee1000) [inline container_pipeline/workers/base.py]: Hmm ok i see

Comment (bamachrn) [inline container_pipeline/workers/base.py]: yes agree, let the CI be green

**Comment by bamachrn** [comment]
#dotests

File: container_pipeline/workers/base.py [hunk 2]
```diff
@@ -46,23 +48,36 @@ def export_logs(self, logs, destination):
...
+
         while True:
             try:
                 job_obj = None
+                previous_job_name = current_job_name
+                previous_trig_time = current_trig_time
                 job_obj = self.queue.get()
                 job = json.loads(job_obj.body)
+                current_job = get_job_name(job)
+                current_trig_time = datetime.time
```
Comment (rtnpro) [inline container_pipeline/workers/base.py]: ``datetime.time`` is a function

File: container_pipeline/workers/base.py [hunk 3]
```diff
@@ -46,23 +48,36 @@ def export_logs(self, logs, destination):
 
     def run(self):
         """Run worker"""
+        current_job_name = None
+        previous_job_name = None
+        current_trig_time = datetime.time
+        previous_trig_time = datetime.time
```
Comment (rtnpro) [inline container_pipeline/workers/base.py]: ``datetime.time`` is a function.

File: container_pipeline/workers/base.py [hunk 4]
```diff
@@ -46,23 +48,36 @@ def export_logs(self, logs, destination):
...
                 self.logger.info('Got job: {}'.format(job))
-                try:
-                    self.handle_job(job)
-                except Exception as e:
-                    self.logger.error(
-                        'Error in handling job: {}\nJob details: {}'.format(
-                            e, job), extra={'locals': locals()}, exc_info=True)
+                if((previous_job_name == current_job_name) and ((current_trig_time - previous_trig_time) < 1000)):
+                    self.logger.critical(
+                        'Current job is repeating frequently: Job details: {}'.format(job), extra={'locals': locals()}, exc_info=True)
```
Comment (rtnpro) [inline container_pipeline/workers/base.py]: This should not be a critical log, rather ``info`` since we are able to handle this scenario.

Comment (rtnpro) [inline container_pipeline/workers/base.py]: You will end up flooding the log file here if the above situation arises.

File: container_pipeline/workers/base.py [hunk 5]
```diff
@@ -46,20 +48,41 @@ def export_logs(self, logs, destination):
...
-                job['logs_dir'], settings.SERVICE_LOGFILE)
-            # Run dfh.clean() to clean log files if no error is encountered in
-            # post delivering build report mails to user
-            dfh = DynamicFileHandler(self.logger, debug_logs_file)
-            self.logger.info('Got job: {}'.format(job))
             try:
-                self.handle_job(job)
+                job_obj = None
+                previous_job_name = current_job_name
+                previous_trig_time = current_trig_time
```
Comment (navidshaikh) [inline container_pipeline/workers/base.py]: can we take these unnecessary operations out of try block ? We had problem earlier in the code where there were long `try` block code snippets. Lets make sure we not going there again.

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by bamachrn** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Comment by rtnpro** [comment]
#dotests-debug

**Pull request merged**